Expanded the _validate_anthropic_response function to include comprehensive checks

S1LV3RJ1NX opened the pull request:
✅ Test

## Changes
- Updated unit tests to utilize the new validation function for verifying response integrity in both non-streaming and streaming scenarios.

S1LV3RJ1NX committed:
- Expanded the _validate_anthropic_response function to include comprehensive checks for response structure, ensuring all required fields are validated.
- Updated unit tests to utilize the new validation function for verifying response integrity in both non-streaming and streaming scenarios.

ishaan-jaff commented on tests/pass_through_unit_tests/test_anthropic_messages_passthrough.py:
```diff
@@ -125,9 +159,36 @@ async def test_anthropic_messages_streaming():
             print("chunk=", chunk)
+            collected_chunks.append(chunk)
+            if "content" in chunk and len(chunk["content"]) > 0:
+                for content_block in chunk["content"]:
```
please can you validate the streaming chunks received

S1LV3RJ1NX replied:
Done, also using Anthropic SDK for non-streaming validation.

ishaan-jaff reviewed (changes requested):
mostly good, missing validation for streaming chunks. please add that

krrishdholakia commented:
One approach - use anthropic sdk's pydantic object and use `.model_validate()` to confirm output matches expected format

S1LV3RJ1NX committed:
Update poetry.lock and pyproject.toml for Anthropic library integration

- Upgraded Poetry version in poetry.lock to 2.0.1.
- Added the Anthropic library (version 0.50.0) to pyproject.toml.
- Enhanced unit tests for Anthropic messages to utilize the new validation method and updated model references for consistency.
- Improved response validation in tests to ensure compliance with the Anthropic Messages API structure.

S1LV3RJ1NX commented:
Testcase:
![image](https://github.com/user-attachments/assets/94dd1291-3b41-4846-9a89-21cc5b3d830a)

S1LV3RJ1NX committed:
Merge branch 'main' of https://github.com/BerriAI/litellm into litellm_messages_support_update_tc

ishaan-jaff commented on pyproject.toml:
```diff
@@ -56,6 +56,7 @@ boto3 = {version = "1.34.34", optional = true}
 redisvl = {version = "^0.4.1", optional = true, markers = "python_version >= '3.9' and python_version < '3.14'"}
 mcp = {version = "1.5.0", optional = true, python = ">=3.10"}
 litellm-proxy-extras = {version = "0.1.14", optional = true}
+anthropic = "^0.50.0"
```
please don't add anthropic as a core dep on litellm 

if you need this in your test you can add it on .circle/config.yaml

S1LV3RJ1NX replied:
The reason this was added to use anthropic SDK's pydantic to directly validate the response. @krrishdholakia had suggested it. Let me know if we still want to use. If not, I can compare the pydantic keys myself.

ishaan-jaff replied:
I understand. You can use it in your test but please don't add this as a core dependency to the litellm pip package.

ishaan-jaff commented on tests/pass_through_unit_tests/test_anthropic_messages_passthrough.py:
```diff
@@ -119,15 +118,71 @@ async def test_anthropic_messages_streaming():
         async for chunk in response:
             print("chunk=", chunk)
+            collected_chunks.append(chunk)
+            
```
is there no way to validate the chunks the same way you did for streaming @S1LV3RJ1NX ?

S1LV3RJ1NX replied:
do you mean non-streaming? I could not find one.

ishaan-jaff reviewed (changes requested):
reviewed

S1LV3RJ1NX committed:
Remove Anthropic library from poetry.lock and pyproject.toml, and update CircleCI configuration to install the library during testing.

S1LV3RJ1NX committed:
Merge remote-tracking branch 'origin/main' into litellm_messages_support_update_tc

S1LV3RJ1NX committed:
Merge branch 'main' of https://github.com/BerriAI/litellm into litellm_messages_support_update_tc
